vendor/bat-native-ads/src/bat/ads/internal/rpill/rpill_helper.cc

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/rpill/rpill_helper.h"
+
+#include <string>
+#include <vector>
+
+#include "base/strings/string_util.h"
+#include "bat/ads/ads.h"
+
+namespace ads {
+
+namespace {
+
+std::string GetHardware() {
+  const std::string manufacturer = base::ToLowerASCII(_sys_info.manufacturer);
+  const std::string model = base::ToLowerASCII(_sys_info.model);
+  return manufacturer + model;
+}
+
+std::vector<std::string> GetKeywords() {
+  return {
+    "amazon",
+    "virtualbox",
+    "vmware",
+    "xen"
+  };
+}
+
+bool IsVirtualMachine() {
+  const std::string hardware = GetHardware();
+  const std::vector<std::string> keywords = GetKeywords();
+
+  for (const auto& keyword : keywords) {
+    if (hardware.find(keyword) != std::string::npos) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+}  // namespace
+
+RPillHelper* g_rpill_helper_for_testing = nullptr;
+
+RPillHelper::RPillHelper() = default;
+
+RPillHelper::~RPillHelper() = default;
+
+void RPillHelper::set_for_testing(
+    RPillHelper* rpill_helper) {
+  g_rpill_helper_for_testing = rpill_helper;
+}
+
+bool RPillHelper::IsUncertainFuture() const {
+  return IsVirtualMachine();
+}
+
+RPillHelper* RPillHelper::GetInstance() {
+  if (g_rpill_helper_for_testing) {
```

Comment:
as in the other PR, why would we ever have to mock this in testing now? Also doesn't seem like it needs to be an instance at all

Reply:
Agreed as this PR removed the need to mock RPill :-)

Reply:
Fixed